core/src/OAuth2/OAuth2Client.cs

```diff
@@ -166,7 +167,26 @@ public static void CreateErrorResponse(HttpResponse response, RequestContext req
                     ex);
             }
 
-            requestContext.Logger.ErrorPii(serviceEx);
+            // For device code flow, AuthorizationPending can occur a lot while waiting
+            // for the user to auth via browser and this causes a lot of error noise in the logs.
+            // So suppress this particular case to an Info so we still see the data but don't 
+            // log it as an error since it's expected behavior while waiting for the user.
+            bool shouldLogError = true;
+            if (serviceEx is CoreException coreException)
+            {
+                if (string.Compare(coreException.ErrorCode, OAuth2Error.AuthorizationPending,
+                        StringComparison.OrdinalIgnoreCase) == 0)
+                {
+                    requestContext.Logger.InfoPii(serviceEx);
+                    shouldLogError = false;
+                }
+            }
+
+            if (shouldLogError)
+            {
+                requestContext.Logger.ErrorPii(serviceEx);
+            }
```

Comment:
my suggestion would be to limit the fix to this.. using the existing exceptions. The ExceptionFactory might be the way to filter on the type. Not sure that the change for ADAL is possible as it will be breaking customers.